tkg/client/upgrade_cluster.go

```diff
@@ -1191,8 +1198,41 @@ func (c *TkgClient) handleKappControllerUpgrade(regionalClusterClient, currentCl
 			return errors.Wrapf(err, "unable to check if Cluster API Provider for AWS is enabled")
 		}
 	} else {
-		if err := regionalClusterClient.UpdateAWSCNIIngressRules(upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace); err != nil {
-			return errors.Wrapf(err, "unable to update AWS CNI ingress rules")
+		kappControllerAWSIngressRule := capav1beta2.CNIIngressRule{
+			Description: "kapp-controller",
+			Protocol:    capav1beta2.SecurityGroupProtocolTCP,
+			FromPort:    DefaultKappControllerHostPort,
+			ToPort:      DefaultKappControllerHostPort,
+		}
+		if err := regionalClusterClient.UpdateAWSCNIIngressRules(upgradeClusterConfig.ClusterName, upgradeClusterConfig.ClusterNamespace, kappControllerAWSIngressRule); err != nil {
+			return errors.Wrapf(err, "unable to update AWS CNI ingress rules for %s/%s kapp-controller", upgradeClusterConfig.ClusterNamespace, upgradeClusterConfig.ClusterName)
+		}
+	}
+
+	return nil
+}
+
+// PrepareAddonsManagerUpgrade is only meant for testing prepareAddonsManagerUpgrade
+func (c *TkgClient) PrepareAddonsManagerUpgrade(regionalClusterClient clusterclient.Client, upgradeClusterConfig *ClusterUpgradeInfo) error {
+	return c.prepareAddonsManagerUpgrade(regionalClusterClient, upgradeClusterConfig)
+}
+
+func (c *TkgClient) prepareAddonsManagerUpgrade(regionalClusterClient clusterclient.Client, upgradeClusterConfig *ClusterUpgradeInfo) error {
+	// Update AWSCluster cniIngressRules to include addon-manager port only if CAPA is running on the management cluster
+	if err := regionalClusterClient.GetResource(&corev1.Namespace{}, clusterclient.CAPAControllerNamespace, clusterclient.CAPAControllerNamespace, nil, nil); err != nil {
+		// if capa-system namespace doesn't exist, then assume that updates to AWSCluster are not required during upgrade.
```

Comment:
Should we call out that loud in a log

Reply:
No, i don't think so, or only at a high debug level

Reply:
If we  do, we'll see it allt he time in vcenter and that will confuse the user pretty bad I think. 


> Hnm, expect BYO VPC from now on, so we expect these rules to be set up by the customer up front. Furthermore, we have many customers, perhaps even the majority, who do not have access to modify security group rules.

^^ that is bad news all the way around not just for this.  Are you sure that is the case... I have never heard of anyone using aws without having access to security groups. It would make aws impossible to use no?
If customers do not have access to their security rules that will basically make it impossible to deploy on aws...

Reply:
If we do, we'll see it allt he time in vcenter and that will confuse the user pretty bad I think.

Hnm, expect BYO VPC from now on, so we expect these rules to be set up by the customer up front. Furthermore, we have many customers, perhaps even the majority, who do not have access to modify security group rules.

^^ that is bad news all the way around not just for this. Are you sure that is the case... I have never heard of anyone using aws without having access to security groups. It would make aws impossible to use no?
If customers do not have access to their security rules that will basically make it impossible to deploy on aws...

Reply:
But if that is the case, we don't need this patch at all and instead we dump the responsability on the user to setup the correct security group rules. but I have no idea how that would work.